File: VB/CustomDictionarySample.Web/App_Code/CustomDictionaryService.vb
Imports Microsoft.VisualBasic
Imports System.Linq
Imports System.ServiceModel.Activation

<AspNetCompatibilityRequirements(RequirementsMode := AspNetCompatibilityRequirementsMode.Allowed)> _
Public Class CustomDictionaryService
	Implements ICustomDictionaryService
	Private Const customDictionaryPath As String = "~/App_Data/Dictionaries/CustomWords.txt"

	#Region "ICustomDictionaryService Members"

	Public Function LoadCustomDictionary() As String() Implements ICustomDictionaryService.LoadCustomDictionary
		Dim path As String = System.Web.HttpContext.Current.Server.MapPath(customDictionaryPath)
		Return System.IO.File.ReadLines(path).ToArray()
	End Function

	Public Sub SaveCustomDictionary(ByVal words() As String) Implements ICustomDictionaryService.SaveCustomDictionary
		Dim path As String = System.Web.HttpContext.Current.Server.MapPath(customDictionaryPath)
		System.IO.File.WriteAllLines(path, words)
	End Sub

	#End Region
End Class

File: CS/CustomDictionarySample.Web/App_Code/CustomDictionaryService.cs
using System.Linq;
using System.ServiceModel.Activation;

[AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
public class CustomDictionaryService : ICustomDictionaryService {
    private const string customDictionaryPath = "~/App_Data/Dictionaries/CustomWords.txt";

    #region ICustomDictionaryService Members

    public string[] LoadCustomDictionary() {
        string path = System.Web.HttpContext.Current.Server.MapPath(customDictionaryPath);
        return System.IO.File.ReadLines(path).ToArray();
    }

    public void SaveCustomDictionary(string[] words) {
        string path = System.Web.HttpContext.Current.Server.MapPath(customDictionaryPath);
        System.IO.File.WriteAllLines(path, words);
    }

    #endregion
}
